internal/resources/iampolicy/scope_helper_test.go

```diff
@@ -0,0 +1,245 @@
+			actualScope, err := getScopeType(test.input)
+			if test.expErr != nil {
+				require.EqualError(t, err, test.expErr.Error())
+				return
+			}
+
+			require.NoError(t, err)
+			require.EqualValues(t, test.expScope, actualScope)
+
+			fn := constructScopeFullName(actualScope, test.input)
+			require.EqualValues(t, test.expFullName, fn)
+		})
+	}
+}
+
+func getSchemaData(t *testing.T, opts ...resourceDataOpt) *schema.ResourceData {
+	d := new(schema.Resource)
+	d.Schema = iamPolicySchema
+
+	rd := d.TestResourceData()
+	scopeData := map[string][]interface{}{}
+
+	for _, o := range opts {
+		o(scopeData)
+	}
+
+	err := rd.Set(scopeKey, []interface{}{scopeData})
+	require.NoError(t, err)
+
+	return rd
+}
+
+func withClusterGroup(val string) resourceDataOpt {
+	return func(res map[string][]interface{}) {
+		res[clusterGroupKey] = []interface{}{
+			map[string]string{
+				clusterGroupNameKey: val,
+			},
+		}
+	}
+}
+
+func withOrg(val string) resourceDataOpt {
+	return func(res map[string][]interface{}) {
+		res[organizationKey] = []interface{}{
+			map[string]string{
+				orgIDKey: val,
+			},
+		}
+	}
+}
+
+func withWorkspace(val string) resourceDataOpt {
+	return func(res map[string][]interface{}) {
+		res[workspaceKey] = []interface{}{
+			map[string]string{
+				workspaceNameKey: val,
+			},
+		}
+	}
+}
+
+func withCluster(mc, prov, name string) resourceDataOpt {
+	return func(res map[string][]interface{}) {
+		res[clusterKey] = []interface{}{
+			map[string]string{
+				managementClusterNameKey: mc,
+				provisionerNameKey:       prov,
+				clusterNameKey:           name,
+			},
+		}
+	}
+}
+
+func withNamespace(mc, prov, cluster, ns string) resourceDataOpt {
+	return func(res map[string][]interface{}) {
+		res[namespaceKey] = []interface{}{
+			map[string]string{
+				managementClusterNameKey: mc,
+				provisionerNameKey:       prov,
+				clusterNameKey:           cluster,
+				namespaceNameKey:         ns,
+			},
+		}
+	}
+}
+
+func TestGetRoleBindingSchemaData(t *testing.T) {
+	d := new(schema.Resource)
+	d.Schema = iamPolicySchema
+
+	rd := d.TestResourceData()
+
+	err := rd.Set(
+		roleBindingsKey,
+		[]interface{}{
+			map[string]interface{}{
+				roleKey: "dummy-role",
+				subjectsKey: []interface{}{
+					map[string]interface{}{
+						subjectNameKey: "sub",
+						subjectKindKey: "kind",
+					},
+					map[string]interface{}{
+						subjectNameKey: "sub2",
+						subjectKindKey: "kind2",
+					},
+				},
+			},
+			map[string]interface{}{
+				roleKey: "dummy-role-2",
+				subjectsKey: []interface{}{
+					map[string]interface{}{
+						subjectNameKey: "sub",
+						subjectKindKey: "kind",
+					},
+					map[string]interface{}{
+						subjectNameKey: "sub2",
+						subjectKindKey: "GROUP",
+					},
+				},
+			},
+		})
+	require.NoError(t, err)
+
+	got := constructRoleBindingList(rd)
+
+	fmt.Println(len(got))
+
+	for _, each := range got {
+		for _, s := range each.Subjects {
+			fmt.Println(each.Role, s.Name, *s.Kind)
+		}
+	}
```

Comment:
This is good but can we add `t.Run(test.name, func(t *testing.T){}` instead of `fmt.Println()`?